src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/ApiMapComponent.java

```diff
@@ -0,0 +1,45 @@
+package io.stargate.sgv2.jsonapi.api.model.command.table;
+
+import io.stargate.sgv2.jsonapi.config.constants.TableDescConstants;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Enum for API map component representation. It could be used in createIndex for map column,
+ * filtering on map column.
+ */
+public enum ApiMapComponent {
+  KEYS(TableDescConstants.MapTypeComponent.keys),
+  VALUES(TableDescConstants.MapTypeComponent.values);
+
+  private final String apiName;
+
+  private static final Map<String, ApiMapComponent> COMPONENT_MAP;
+
+  static {
+    COMPONENT_MAP = new HashMap<>();
+    for (ApiMapComponent component : ApiMapComponent.values()) {
+      COMPONENT_MAP.put(component.apiName.toLowerCase(), component);
+    }
+  }
+
+  public static Optional<ApiMapComponent> fromUserInput(String userInput) {
+    if (userInput == null) {
+      return Optional.empty();
+    }
+    var lowerCaseInput = userInput.toLowerCase();
+    if (!COMPONENT_MAP.containsKey(lowerCaseInput)) {
+      return Optional.empty();
+    }
+    return Optional.of(COMPONENT_MAP.get(lowerCaseInput));
```

Comment:
just a thought, Optional.ofNullable() can make code this like easier, it can return an empty Optional or one with a value it in